addons/product/report/product_pricelist.py

```diff
@@ -2,101 +2,78 @@
-                val = {
-                     'id':product['id'],
-                     'name':product['name'],
-                     'code':product['code']
-                }
-                i = 1
-                for qty in self.quantity:
-                    if qty == 0:
-                        val['qty'+str(i)] = 0.0
+            val = {
+                'name': product.name,
+                'code': product.code,
+                'attribute': ', '.join([attr.name for attr in product.attribute_value_ids])
+            }
+            for c, qty in enumerate(self.quantity, start=1):
+                field_name = 'qty%s' % str(c)
+                if qty > 0:
+                    price_dict = self.pricelist.price_get(product.id, qty)
+                    if price_dict[self.pricelist.id]:
+                        price = price_dict[self.pricelist.id]
                     else:
-                        val['qty'+str(i)]=self._get_price(self.pricelist, product['id'], qty)
-                    i += 1
-                products.append(val)
-            res.append({'name':cat[1],'products': products})
+                        price = product.list_price
+                    val[field_name] = price
+            res.append(val)
         return res
 
-    def _get_price(self, pricelist_id, product_id, qty):
-        sale_price_digits = self.get_digits(dp='Product Price')
-        pricelist = self.pool.get('product.pricelist').browse(self.cr, self.uid, [pricelist_id], context=self.localcontext)[0]
-        price_dict = self.pool.get('product.pricelist').price_get(self.cr, self.uid, [pricelist_id], product_id, qty, context=self.localcontext)
-        if price_dict[pricelist_id]:
-            price = self.formatLang(price_dict[pricelist_id], digits=sale_price_digits, currency_obj=pricelist.currency_id)
-        else:
-            res = self.pool.get('product.product').read(self.cr, self.uid, [product_id])
-            price =  self.formatLang(res[0]['list_price'], digits=sale_price_digits, currency_obj=pricelist.currency_id)
-        return price
-
-
-class report_product_pricelist(osv.AbstractModel):
-    _name = 'report.product.report_pricelist'
-    _inherit = 'report.abstract_report'
-    _template = 'product.report_pricelist'
-    _wrapped_report_class = product_pricelist
+    @api.multi
+    def render_html(self, data):
+        Report = self.env['report']
+        form = data.get('form')
+        self.model = self.env.context.get('active_model')
+        self._set_quantity(form)
+        self.pricelist = self.env['product.pricelist'].browse(form['price_list'])
+        selected_records = self.env[self.model].browse(data.get('ids'))
+        docargs = {
+            'doc_ids': data.get('ids'),
+            'doc_model': self.model,
+            'docs': selected_records,
+            'time': time,
+            'get_pricelist': self.pricelist.name,
```

Comment:
do not add prefix 'get_'. it's value not a method. so just pricelist, currency, titles, categories enough :)